### DISCUSSION TITLE
How to mark a enterprise repository as internal using GitHub rest API?

### DISCUSSION THREAD

**Discussion by syedshariq-a** [community/community#58073] [Enterprise] [answered]
### Select Topic Area

Question

### Body

Hi, I am working with organization repositories (GitHub Enterprise) and want to mark repos as internal using API. 

I know there is a `private` boolean flag that I can use to mark the repos as private and public in rest apis, but how do I mark a repo as internal using the API? I wasn't able to find a reference to this hence asking here.

I have repos created as private and I want to convert them to internal using rest APIs.

Let's say the organization name is `foo` and private org repo name is `bar`, how would I go about marking this private repo as internal using the REST API?

I will be using a GitHub App with admin permissions on the organization to perform the REST API calls.

**Comment by nguyenngoclongdev** [comment, marked as answer]
To mark an enterprise repository as internal using the GitHub REST API, you can follow these steps:
1. Authenticate with GitHub using an access token that has the necessary permissions to modify the repository. You should include the access token in an Authorization header when making requests to the API.
2. Make a PATCH request to the following endpoint, replacing ENTERPRISE_NAME with the name of your enterprise, REPO_OWNER with the owner of the repository, and REPO_NAME with the name of the repository: `PATCH https://api.github.com/repos/ENTERPRISE_NAME/REPO_OWNER/REPO_NAME`
3. In the request body, include a JSON object with the following key-value pair: `{ "visibility": "internal" }`
4. Send the PATCH request to the API endpoint, including the Authorization header with your access token.

With curl:
`curl -X PATCH -H "Authorization: token YOUR_ACCESS_TOKEN" -d '{"visibility": "internal"}' https://api.github.com/repos/ENTERPRISE_NAME/REPO_OWNER/REPO_NAME`

**Reply by calebAtIspot** [reply]
note you can specify visibility internal with the create repo API call as well
